### PR TITLE
fix(torrent): `get_files()` should throw `KeyError` if `files` not fetched

### PR DESCRIPTION
**Pull request by dechamps**
BREAKING CHANGE: `torrent.get_files()` will raise a KeyError if field `files` not fetched. `file.priority` and `file.selected` are not optional, based on fields fetched.

### CODE REVIEW

**Comment by trim21** [comment]
I'm considering this https://github.com/trim21/transmission-rpc/pull/422 

This method should raise KeyError if `files` not fetched.

**Comment by dechamps** [comment]
> I'm considering this #422

I commented on that PR - I don't think it makes sense to force users to fetch fields they don't need.

> This method should raise KeyError if `files` not fetched.

I agree that would make sense. Would you like me to extend this PR to also raise `KeyError` if `files` is not present?

**Comment by trim21** [comment]
> > I'm considering this #422
> 
> I commented on that PR - I don't think it makes sense to force users to fetch fields they don't need.
> 
> > This method should raise KeyError if `files` not fetched.
> 
> I agree that would make sense. Would you like me to extend this PR to also raise `KeyError` if `files` is not present?

both sounds reasonable to me. But I plan to make breaking change when tr 4.1.0 is release. 

Would mind update PR and wait until tr 4.1.0 is released? I'll merge it then.

**Comment by dechamps** [comment]
> I plan to make breaking change when tr 4.1.0 is release

I'm not sure why you see this as a breaking change? With the previous code the only way to get `get_files()` to work and give the correct output was to fetch `files`, `properties` and `wanted`. This PR does not change the behavior in that case. It only changes the behavior for usage patterns where the previous code did not work at all in the first place. Can't break what's already broken.

> Would mind update PR

I have updated the PR to make `get_files()` raise `KeyError` if `files` was not fetched.

File: transmission_rpc/torrent.py [hunk 1]
```diff
@@ -429,26 +429,21 @@ def get_files(self) -> list[File]:
...
-                    id=id,
-                )
-                for id, file, raw_priority, raw_selected in zip(indices, files, priorities, wanted)
+        files = self.fields["files"]
+        indices = range(len(files))
+        priorities = self.fields.get("priorities", [None] * len(files))
+        wanted = self.fields.get("wanted", [None] * len(files))
+        return [
+            File(
+                selected=bool(raw_selected) if raw_selected is not None else None,
```
Comment (trim21) [inline transmission_rpc/torrent.py:438]: can we just handle this before loop…

Comment (dechamps) [reply transmission_rpc/torrent.py:438]: I'm not sure what you mean?

**Comment by dechamps** [comment]
Adjusted the PR to account for CI failures, which I didn't notice until now because it didn't run on my development branch (see #448 about that).

File: transmission_rpc/torrent.py (hunk 1 above)
Comment (trim21) [reply transmission_rpc/torrent.py:438]:
```python
if 'priorities' in self.fields:
    priorities = (... for v in self.fields['priorities'])
else:
    priorities = [None] * len(files)
```

Comment (dechamps) [reply transmission_rpc/torrent.py]: Done.

**Comment by trim21** [comment]
> > I plan to make breaking change when tr 4.1.0 is release
> 
> I'm not sure why you see this as a breaking change? With the previous code the only way to get `get_files()` to work and give the correct output was to fetch `files`, `properties` and `wanted`. This PR does not change the behavior in that case. It only changes the behavior for usage patterns where the previous code did not work at all in the first place. Can't break what's already broken.


obviously it will throw `KeyError` now

**Comment by dechamps** [comment]
> obviously it will throw `KeyError` now

It will not throw `KeyError` in code that is not already broken.

If a caller called this method without fetching `files` first, the previous code would return an incorrect result (i.e. an empty list). Meaning it's already broken.

The new code takes this already broken case and breaks it in a different, clearer way, by raising an error.

Caller code that is correct (i.e. callers who fetch `files` before calling `get_files()`) will not be broken by this change. The change only affects callers that are already broken in the first place.

**Pull request merged**